editoast/src/views/train_schedule/simulation_report.rs

```diff
@@ -358,7 +358,8 @@ fn project_head_positions(
             let loc = &train_locations[loc_index];
             let position =
                 projection.track_position(&Identifier(loc.track_section.clone()), loc.offset);
-            assert!(position.is_some());
+            let last_position = current_curve.last().unwrap().position;
+            assert!(position.is_some_and(|p| p > last_position));
```

Comment:
Why don't we use [`debug_assert!()`](https://doc.rust-lang.org/std/macro.debug_assert.html) instead?

Reply:
It's because we want the prod to fail. We have a hard time reproducing a bug that we know is there. So we want the assert to fail in production in hope to better understand what causes it, since we'll have an immediate feedback.